inc/classes/subscriber/third-party/plugins/class-mobile-subscriber.php

```diff
@@ -0,0 +1,334 @@
+<?php
+namespace WP_Rocket\Subscriber\Third_Party\Plugins;
+
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+defined( 'ABSPATH' ) || die( 'Cheatin&#8217; uh?' );
+
+/**
+ * Class that handles events related to plugins that add mobile themes.
+ *
+ * @since  3.1.4
+ * @author Grégory Viguier
+ */
+class Mobile_Subscriber implements Subscriber_Interface {
+
+	/**
+	 * Options to activate when a mobile plugin is active.
+	 *
+	 * @since  3.1.4
+	 * @access protected
+	 * @author Grégory Viguier
+	 *
+	 * @var array
+	 */
+	protected static $options = [
+		'cache_mobile'            => 1,
+		'do_caching_mobile_files' => 1,
+	];
+
+	/**
+	 * Cache the value of self::is_mobile_plugin_active().
+	 *
+	 * @since  3.1.4
+	 * @access protected
+	 * @author Grégory Viguier
+	 *
+	 * @var array An array of arrays of booleans.
+	 *            First level of keys corresponds to the network ID. Second level of keys corresponds to the blog ID.
+	 */
+	protected static $is_mobile_active = [];
+
+	/**
+	 * Returns an array of events that this subscriber wants to listen to.
+	 *
+	 * @since  3.1.4
+	 * @access public
+	 * @author Grégory Viguier
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		// In case a mobile plugin has already been activated.
+		$do   = [];
+		$undo = [];
+
+		foreach ( static::get_mobile_plugins() as $plugin ) {
+			if ( \did_action( 'activate_' . $plugin ) ) {
+				$do[] = $plugin;
+			}
+			if ( \did_action( 'deactivate_' . $plugin ) ) {
+				$undo[] = $plugin;
+			}
+		}
+
+		if ( array_diff( $do, $undo ) || array_diff( $undo, $do ) ) {
+			static::update_mobile_cache_activation();
+		}
+
+		// Register events.
+		$events = [
+			// Plugin activation/deactivation.
+			'add_option_active_plugins'                  => [ 'add_option_callback', 10, 2 ],
+			'update_option_active_plugins'               => [ 'update_option_callback', 10, 2 ],
```

Comment:
Side note:
I use those hooks instead of `"activate_{$plugin}"` and `"deactivate_{$plugin}"` because of [deactivate_plugins()](https://wpseek.com/function/deactivate_plugins/). At the end of the function we can see that when the hook `"deactivate_{$plugin}"` is triggered, the option `active_plugins/active_sitewide_plugins` is not updated yet: that means that `is_plugin_active()` will return an incorrect result when testing for active mobile plugins on config file update.
With [activate_plugin()](https://wpseek.com/function/activate_plugin/) I could have used the hook `activated_plugin`, which is triggered AFTER the option is updated, but with the hook `"update_option_{$option}"` we can handle both activation and deactivation. Moreover, both activation and deactivation lead to the same action.
As about the `add_option_` and `delete_option_`, I'm just being Greg ¯\_(ツ)_/¯